test/unit/spec/Registration_spec.js

```diff
@@ -642,14 +692,49 @@ function (Q, _, $, OktaAuth, Backbone, Util, Expect, Beacon, RegForm, RegSchema,
         return setup(setting)
         .then(function (test) {
           $.ajax.calls.reset();
-          test.form.setEmail('test@example.com');
+          test.form.setUserName('test@example.com');
+          test.form.setPassword('Abcd1234');
+          test.form.setFirstname('firstName');
+          test.form.submit();
+          var model = test.router.controller.model;
+          spyOn(Backbone.Model.prototype, 'save').and.returnValue($.Deferred().resolve());
+          model.save();
+          expect(setting.registration.postSubmit).toHaveBeenCalled();
+        });
+      });
+      itp('calls postSubmit call onSuccess assert username is same as email', function () {
+        var parseSchemaSpy = jasmine.createSpy('parseSchemaSpy');
+        var preSubmitSpy = jasmine.createSpy('preSubmitSpy');
+        var setting = {
+          'registration': {
+            'parseSchema': function (resp, onSuccess, onFailure) {
+              parseSchemaSpy(resp, onSuccess, onFailure);
+              onSuccess(resp);
+            },
+            'preSubmit': function (postData, onSuccess, onFailure) {
+              preSubmitSpy(postData, onSuccess, onFailure);
+              onSuccess(postData);
+            },
+            'postSubmit': jasmine.createSpy('postSubmitSpy')
+          }
+        };
+        return setup(setting)
+        .then(function (test) {
+          $.ajax.calls.reset();
+          test.form.setUserName('test@example.com');
           test.form.setPassword('Abcd1234');
           test.form.setFirstname('firstName');
           test.form.submit();
           var model = test.router.controller.model;
           spyOn(Backbone.Model.prototype, 'save').and.returnValue($.Deferred().resolve());
           model.save();
+          test.router.navigate('signin/register-complete', {trigger: true});
           expect(setting.registration.postSubmit).toHaveBeenCalled();
+          expect(model.get('userName')).toBe('test@example.com');
```

Comment:
Not sure if this expect `expect(model.get('userName')).toBe('test@example.com');` is adding any value, since we set the userName and just validate that its set ?